### DISCUSSION TITLE
help with collision

### DISCUSSION THREAD

**Discussion by fernandopabst** [Annoraaq/grid-engine#139] [Q&A] [answered]
Hi annoraaq,

First of all, this is brilliant work and you have been very kind to provide comprehensive documentation alongside. I've spent a good few hours this weekend going through everything and checking out some examples, but I am afraid to say I haven't been able to nail down the collision system. I am just starting my coding journey so I appreciate this may be a very silly question, but would you be able to pinpoint in the sandbox below where it is that I am missing something? I have quite a few things commented out where I've explore some options, but my player will still only collide with the npc. I can also see that both sprites are on the top layer, above everything else, so that could explain the issues; however, I've been unable to hardcode the layering through any of the methods that I've found online.

My idea is to build a small game to help language learners learn Welsh, and your plugin has been a massive help towards that goal.

This is the sandbox with my experimentations:

https://codesandbox.io/s/hungry-sun-ykb6f?file=/src/index.js

Thank you again!
Fernando

**Comment by Annoraaq** [comment, marked as answer]
Hi Fernando,

I am happy to hear you like the plugin.

I think I found your issue. You are using `collides` as a tile property but you are not setting this in the configuration. It used to work with versions < 2.0.0  of GridEngine. But since version 2, `ge_collide` is the default name for the collision property. You now have 2 ways to solve your problem:

1. You use Tiled and change the name of the collision property from `collides` to `ge_collide` in all your tilemaps.

2. You add `collisionTilePropertyName: 'collides'` to your GridEngine config: https://annoraaq.github.io/grid-engine/api/config.html

I think the documentation might be a bit confusing about this because 

1. the screenshot [here](https://annoraaq.github.io/grid-engine/usage/collision.html) is out of date. The default name of the property changed from `collides` to `ge_collide` in version 2. I will update this, so it is easier to understand.

2. `collisionTilePropertyName` is not described very well in the docs

3. `collisionTilePropertyName` is not mentioned in the collision section of the docs

Let me know if this helps. And thank you for pointing out a weakness in the docs!

**Reply by fernandopabst** [reply]
That worked perfectly! I added the `collisionTilePropertyName` property to the gridEngineConfig, and it instantly worked. Thank you so much for your swift reply! I will now do some work to add some interactivity and small , quest-like scenes with a language learning outcome in mind - will make sure to give you credit for the plugin and tutorial if it ever goes live to my students.

And your docs are great, really clearly laid out, including the tutorial. Comprehensive and to-the-point.

Thanks again

**Reply by Annoraaq** [reply]
I am glad I could help. Feel free to share your project. I am planning to show a list of projects/games using the plugin in the readme.
